code-studio/src/js/pd/workshop_dashboard/components/workshop_form.jsx

```diff
@@ -174,8 +174,8 @@ var WorkshopForm = React.createClass({
       destroyedSessions.push(removedSession);
     }
     this.setState({
-      sessions: sessions,
-      destroyedSessions: destroyedSessions
+      sessions: _.cloneDeep(sessions),
+      destroyedSessions: _.cloneDeep(destroyedSessions)
```

Comment:
I think the cloning for sessions needs to happen sooner (i.e. before we mutate it), otherwise we're still changing the props of the caller.

I don't think we need to clone destroyedSessions at all as this is just a local object.

i.e. I would expect:
```
handleSessionsChange: function (sessions, removedSession) {
    sessions = _.cloneDeep(sessions); 
    sessions.sessionsModified = true;
     var destroyedSessions = [];
     if (removedSession && removedSession.id) {
        destroyedSessions.push(removedSession);
      }
      this.setState({
        sessions: sessions,
        destroyedSessions: destroyedSessions
      });
```

At this point if you want you could take advantage of an ES6 feature and simplify the last bit to `this.setState({sessions, destroyedSessions})`